lib/teleterm/services/unifiedresources/unifiedresources.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unifiedresources
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/teleterm/clusters"
+)
+
+func List(ctx context.Context, cluster *clusters.Cluster, client Client, req *proto.ListUnifiedResourcesRequest) (*ListResponse, error) {
+	unifiedResourcesResponse, err := client.ListUnifiedResources(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	response := &ListResponse{
+		NextKey: unifiedResourcesResponse.NextKey,
+	}
+
+	for _, unifiedResource := range unifiedResourcesResponse.Resources {
+		switch e := unifiedResource.GetResource().(type) {
+		case *proto.PaginatedResource_Node:
+			response.Resources = append(response.Resources, CombinedResource{
+				Server: &clusters.Server{
+					URI:    cluster.URI.AppendServer(e.Node.GetName()),
+					Server: e.Node,
+				},
+			})
+		case *proto.PaginatedResource_DatabaseServer:
+			response.Resources = append(response.Resources, CombinedResource{
+				Database: &clusters.Database{
+					URI:      cluster.URI.AppendDB(e.DatabaseServer.GetName()),
```

Comment:
Should this be getting name from the `Database` instead? I forget how the URIs for databases work in Connect and which one is required